timelock-agent/src/main/java/com/palantir/timelock/paxos/PaxosAgent.java

```diff
@@ -0,0 +1,124 @@
+            Consumer<Object> registrar) {
+        this(install, runtime, ImmutableTimeLockDeprecatedConfiguration.builder().build(), registrar);
+    }
+
+    public PaxosAgent(TimeLockInstallConfiguration install,
+            Observable<TimeLockRuntimeConfiguration> runtime,
+            TimeLockDeprecatedConfiguration deprecated,
+            Consumer<Object> registrar) {
+        super(install, runtime, deprecated, registrar);
+        Preconditions.checkState(install.algorithm() instanceof PaxosInstallConfiguration,
+                "Cannot initialize a Paxos agent with a non-Paxos algorithm configuration.");
+        this.paxosInstall = (PaxosInstallConfiguration) install.algorithm();
+
+        this.paxosResource = PaxosResource.create(paxosInstall.dataDirectory().toString());
+        this.leadershipAgent = new PaxosLeadershipCreator(install, runtime, registrar);
+        this.lockCreator = new LockCreator(runtime, deprecated);
+        this.timestampCreator = new PaxosTimestampCreator(paxosResource,
+                PaxosRemotingUtils.getRemoteServerPaths(install),
+                install.cluster().cluster().security().map(SslSocketFactories::createSslSocketFactory),
+                runtime.map(x -> x.algorithm().orElse(ImmutablePaxosRuntimeConfiguration.builder().build())));
+    }
+
+    @Override
+    public void createAndRegisterResources() {
+        registerPaxosResource();
+        registerPaxosExceptionMappers();
+        leadershipAgent.registerLeaderElectionService();
+
+        // Finally, register the endpoints associated with the clients.
+        super.createAndRegisterResources();
+    }
+
+    // No runtime configuration at the moment.
+    private void registerPaxosResource() {
+        registrar.accept(paxosResource);
+    }
+
+    private void registerPaxosExceptionMappers() {
+        registrar.accept(new BlockingTimeoutExceptionMapper());
+        registrar.accept(new NotCurrentLeaderExceptionMapper());
+        registrar.accept(new TooManyRequestsExceptionMapper());
+    }
+
+    @Override
+    public TimeLockServices createInvalidatingTimeLockServices(String client) {
+        ManagedTimestampService timestampService = instrument(
+                ManagedTimestampService.class,
+                leadershipAgent.wrapInLeadershipProxy(
+                        timestampCreator.createPaxosBackedTimestampService(client),
+                        ManagedTimestampService.class),
+                client);
+        RemoteLockService lockService = instrument(
+                RemoteLockService.class,
+                leadershipAgent.wrapInLeadershipProxy(
+                        lockCreator::createThreadPoolingLockService,
+                        RemoteLockService.class),
+                client);
+
+        return TimeLockServices.create(timestampService, lockService, timestampService);
+    }
+
+    @Override
+    protected boolean configurationFilter(TimeLockRuntimeConfiguration runtimeConfiguration) {
```

Comment:
Ah, yeah this reads weirdly without the live reloading functionality. When we live reload, we will filter the observable on this, ignoring configurations that don't match the filter.

The idea is that we don't want to live reload runtime configs that are incompatible with our install config. It seems nontrivial to otherwise enforce invariants _across_ config objects.